overlord/snapstate/check_snap.go

```diff
@@ -351,6 +351,12 @@ func checkBases(st *state.State, snapInfo, curInfo *snap.Info, flags Flags, devi
 	if snapInfo.Base == "" {
 		return nil
 	}
+	if snapInfo.Base == "none" {
+		if len(snapInfo.Hooks) > 0 || len(snapInfo.Apps) > 0 {
+			return fmt.Errorf("base type %q not supported with apps or hooks", snapInfo.Base)
```

Comment:
Should there be a similar check in snap validation code, so that snap pack complains about it too?

Reply:
Fair point, I think validation is actually the correct place, then here we only need a simple check ("none" -> return nil). Done.